tasty-query/shared/src/main/scala/tastyquery/Subtyping.scala

```diff
@@ -294,8 +294,8 @@ private[tastyquery] object Subtyping:
         else
           tycon2.optSymbol match
             case Some(cls2: ClassSymbol) =>
-              // TODO Handle generic tuple <: TupleN
-              level3WithBaseType(tp1, tp2, cls2)
+              (defn.isTupleNClass(cls2) && isSubtype(tp1, defn.GenericTupleTypeOf(tp2.args)))
```

Comment:
potential optimisation: check tp1 is `*:` first before creating the generic tuple type?

Reply:
That's not trivial at all, because we know nothing about `tp1` at this point. It could be a complicated type alias that eventually becomes a `*:`.